### ISSUE TITLE
[XPath] Functions that determine if a given sequence starts with another sequence or ends with another sequence

### ISSUE THREAD

**Issue by dnovatchev** [qt4cg/qtspecs#96]
It is surprising that we are at version 4 and still are missing:

```xq
(1) fn:starts-with-sequence($container as item()*, $maybe-start as item()*, 
                            $compare as function(item(), item()) as xs:boolean := deep-equal)
                            ) as xs:boolean
```

and

```xq
(2) fn:ends-with-sequence($container as item()*, $maybe-end as item()*, 
                          $compare as function(item(), item()) as xs:boolean := deep-equal)
                          ) as xs:boolean
```

(2) above is a shorthand for:

```xq
fn:starts-with-sequence(reverse($container), reverse($maybe-end)) 
```

**Examples**:

`fn:starts-with-sequence(('a', 'b', 'c', 'd'), ('a', 'b'))`   returns `true()`

`fn:starts-with-sequence(('a', 'b', 'c', 'd'), ('a', 'c'))`   returns `false()`

`fn:ends-with-sequence(('a', 'b', 'c', 'd'), ('c', 'd'))`   returns `true()`

`fn:ends-with-sequence(('a', 'b', 'c', 'd'), ('b', 'd'))`   returns `false()`

`('a', 'b', 'c', 'd') => starts-with-sequence(('a', 'b'))` returns `true()`

`('a', 'b', 'c', 'd') => starts-with-sequence(('a', 'c'))` returns `false()`

`('a', 'b', 'c', 'd') => ends-with-sequence(('c', 'd'))` returns `true()`

`('a', 'b', 'c', 'd') => ends-with-sequence(('b', 'd'))` returns `false()`

**One possible implementation**:

```xq
let $starts-with-sequence := function($seq1 as item()*, $seq2 as item()*, $self as function(*))
{
   empty($seq2)
  or
   head($seq1) eq head($seq2) and $self(subsequence($seq1, 2), subsequence($seq2, 2), $self)
}
  return
    $starts-with-sequence(('a', 'b', 'c', 'd'), ('a', 'b', 'c'), $starts-with-sequence)
```

**Comment by michaelhkay** [comment]
Note workarounds such as

`all(for-each-pair($A, $B, my:compare#2))`

**Comment by dnovatchev** [comment]
> all(for-each-pair($A, $B, my:compare#2))

And for `ends-with-sequence()`:

```xq
   all(for-each-pair(reverse($A), reverse($B), my:compare#2))
```

However this is rather challenging to read and understand.

This proposal aims at achieving conceptual integrity, simplicity, readability and easy understandability / convenience

**Comment by michaelhkay** [comment]
The issue can now be closed, PR #222 was accepted on 2022-11-01
https://qt4cg.org/meeting/minutes/2022/11-01.html#pr-sequence-comparison
